magenta-lib/src/main/scala/magenta/graph/Graph.scala

```diff
@@ -10,21 +11,50 @@ case class MidNode[T](value: T) extends Node[T] {
 }
 case object EndNode extends Node[Nothing]
 
+case class EdgeBuilder[+T](from: Node[T], priority: Int) {
+  def ~>[R >: T](to: Node[R]): Edge[R] = Edge(from, to, priority)
+}
+
 case class Edge[+T](from: Node[T], to: Node[T], priority: Int = 1) {
-  def incPriority(n: Int) = this.copy(priority = priority + n)
   def ~=[R >: T](other: Edge[R]) = from == other.from && to == other.to
 }
 
 object Graph {
   def apply[T](edges: Edge[T]*): Graph[T] = Graph(edges.toSet)
 
+  def apply[T](singleton: T): Graph[T] = Graph(StartNode ~> MidNode(singleton), MidNode(singleton) ~> EndNode)
```

Comment:
Yep - that's fair enough. I don't think it ever will, but it would be clearer to abstract that out.